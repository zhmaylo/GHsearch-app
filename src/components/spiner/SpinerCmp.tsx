import React from 'react';
import { View, StyleSheet, Text, ActivityIndicator } from 'react-native';

// spinner component
export const SpinerÐ¡mp = ({spiner_mes}) => {
    return (
        <View style={styles.container}>
            <ActivityIndicator size="large" color="#000000" />
            <Text>{spiner_mes}</Text>
        </View>)
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'column',
        // justifyContent: "center",
        // alignItems: "center",
    },

})
