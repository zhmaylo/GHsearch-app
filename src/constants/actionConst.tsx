/////////////////////////////
//Start. Search section
//
// search string
export const  SEARCH_REQUEST = 'SEARCH_REQUEST';
// 
//End. Search section
/////////////////////////////

/////////////////////////////
//Start. App init section
//
// application initialization flag
export const IS_APP_INIT = 'IS_APP_INIT';
//
//End. App init section
/////////////////////////////

/////////////////////////////
//Start. Spiner section
//
// spiner toggle (off(false)/ on(true))
export const SPINER_TOGGLE = 'SPINER_TOGGLE';
// 
//End. Spiner section
/////////////////////////////



/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////

