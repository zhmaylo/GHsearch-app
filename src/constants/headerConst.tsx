/////////////////////////////
// Start. Header components
//

// title
export const HEADER_TITLE = 'Github users search app';
export const IMAGE_TITLE = require('./../res/header.png');
//
// End.  Header components
/////////////////////////////

