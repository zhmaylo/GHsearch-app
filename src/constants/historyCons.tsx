/////////////////////////////
// Start. Section 'History cmp'
//
// history storage delay  
export const STORAGE_DELAY = 1000;
// End. Section 'History Cmp'
/////////////////////////////

