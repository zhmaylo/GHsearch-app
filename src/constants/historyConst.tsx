/////////////////////////////
// Start. Section 'History cmp'
//
// history storage delay  
export const STORAGE_DELAY = 1000;
// history title
export const SEARCH_HISTORY = 'Search history:';

// End. Section 'History Cmp'
/////////////////////////////

