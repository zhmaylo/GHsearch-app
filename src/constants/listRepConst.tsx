/////////////////////////////
// Start. Section 'List Rep cmp'
//
// 
export const NAME_REP = 'Name: ';
export const LANGUAGE_REP = 'Language: ';
export const DESCRIPTION_REP = 'Description: ';

// End. Section 'List Rep Cmp'
/////////////////////////////

