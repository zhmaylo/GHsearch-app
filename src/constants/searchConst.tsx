/////////////////////////////
// Start. Section 'Search cmp'
//
// search delay  
export const SEARCH_DELAY = 400;

//
// End. Section 'Search Cmp'
/////////////////////////////

