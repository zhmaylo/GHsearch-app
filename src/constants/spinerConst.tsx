/////////////////////////////
// Start. Section 'Spinner image, '
//

//spinner message 
export const SPINER_MES_LOAD = 'Загрузка...';
export const SPINER_MES_CHECK = 'Проверка...';

//
// End. Section 'Footer image'
/////////////////////////////

