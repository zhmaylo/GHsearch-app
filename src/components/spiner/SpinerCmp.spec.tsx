

import renderer from 'react-test-renderer';
import { SPINER_MES_LOAD } from '../../constants/spinerConst';
import { SpinerСmp } from './SpinerCmp'

test('renders correctly', () => {
    
    const tree = renderer.create(
        SpinerСmp ({spiner_mes: SPINER_MES_LOAD}))
        .toJSON();
    expect(tree).toMatchSnapshot();
});



