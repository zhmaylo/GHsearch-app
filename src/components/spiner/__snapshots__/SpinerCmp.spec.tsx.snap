// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#000000"
    hidesWhenStopped={true}
    size="large"
  />
  <Text>
    Загрузка...
  </Text>
</View>
`;
